src/Sulu/Bundle/AdminBundle/ResourceMetadata/FormResourceMetadataProvider.php

```diff
@@ -111,45 +111,51 @@ public function warmUp($cacheDir)
     private function loadResourceMetadata(): void
     {
         foreach ($this->resources as $resourceKey => $resource) {
-            foreach ($this->locales as $locale) {
-                $this->writeResourceMetadataCache(
-                    $resourceKey,
-                    $resource['form'],
-                    $resource['datagrid'],
-                    $locale
-                );
-            }
+            $this->writeResourceMetadataCache(
+                $resourceKey,
+                $resource['form'],
+                $resource['datagrid']
+            );
         }
     }
 
-    private function writeResourceMetadataCache(string $resourceKey, array $forms, string $list, string $locale): void
+    private function writeResourceMetadataCache(string $resourceKey, array $forms, string $list): void
     {
-        $cache = $this->getCache($locale, $resourceKey);
         $fileResources = [];
 
         $children = [];
         $properties = [];
 
+        // load and merge all given forms
         foreach ($forms as $form) {
-            $formFile = $this->kernel->locateResource($form);
+            $formFile = $this->fileLocator->locate($form);
             /** @var FormMetadata $formStructure */
             $formStructure = $this->formXmlLoader->load($formFile, $resourceKey);
 
             $fileResources = [new FileResource($formFile)];
 
-            $children = array_merge($children, $formStructure->getChildren());
-            $properties = array_merge($properties, $formStructure->getProperties());
+            if ($formStructure->getChildren()) {
+                $children = array_merge($children, $formStructure->getChildren());
```

Comment:
Instead of doing the `getChildren` call twice we could put it in a variable before?